src/economy/economy.cpp

```diff
@@ -16,11 +16,32 @@ enum class economy_reason {
 	pop, factory, rgo, artisan, construction, nation, stockpile, overseas_penalty
 };
 
+std::vector<std::vector<float>> demand_by_category = {
+	{ }, { }, { }, { }, { }, { }, { }, { }, { }
+};
+
 void register_demand(sys::state& state, dcon::nation_id n, dcon::commodity_id commodity_type, float amount, economy_reason reason) {
 	state.world.nation_get_real_demand(n, commodity_type) += amount;
+	(demand_by_category[(int)reason])[commodity_type.index()] += amount;
```

Comment:
this sort of double indirection is an unnecessary inefficiency. Instead add an
```
	property{
		name{ demand_by_category }
		type{array{issue_id}{int32_t}}
	}
```
property to the commodity and resize it to 8 as part of the economy setup code, which will be marginally faster